server/src/main/scala/org/http4s/server/middleware/HSTS.scala

```diff
@@ -0,0 +1,27 @@
+package org.http4s
+package server
+package middleware
+
+import org.http4s.headers.`Strict-Transport-Security`
+import fs2._
+import fs2.interop.cats._
+
+import scala.concurrent.duration.FiniteDuration
+import scala.concurrent.duration._
+
+/** [[Middleware]] to add HTTP Strict Transport Security (HSTS) support adding
+  * the Strict Transport Security headers
+  */
+object HSTS {
+
+  def apply(service: HttpService, maxAge: FiniteDuration = 365.days, includeSubDomains: Boolean = true, preload: Boolean = false): HttpService = Service.lift { req =>
+
+    val header = `Strict-Transport-Security`.unsafeFromDuration(maxAge, includeSubDomains, preload)
```

Comment:
I think anything less than days is probably not used in this case. Basically you want to tell the browser not to use http for the next n days/months/years.